### PR TITLE
[Docs] Add example to reimplement stempel analyzer

### LINKED ISSUE #13150: [doc] Stempel: no information how to reimplement the analyzer
_From @merito on June 11, 2015 8:16_

Hi,

I would like to reimplement the analyzer, for example add my synonyms, but there is no available filter list, for official language analyzers you can find it here - [elastic documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lang-analyzer.html#english-analyzer). Could you add this to the docs?

_Copied from original issue: elastic/elasticsearch-analysis-stempel#43_

### PR DESCRIPTION
**Pull request by cbuescher**
Adding an example of how to re-implement the polish `stempel` analyzer
in case a user want to modify or extend it. In order for the analyzer to be
able to use polish stopwords, also registering a `polish_stop` filter for the 
stempel plugin.

Closes #13150

### CODE REVIEW

**Review by mayya-sharipova — APPROVED** [review]
Thanks @cbuescher, nicely written

File: plugins/analysis-stempel/src/main/java/org/elasticsearch/index/analysis/pl/PolishStopTokenFilterFactory.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.analysis.AbstractTokenFilterFactory;
+import org.elasticsearch.index.analysis.Analysis;
+
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singletonMap;
+
+public class PolishStopTokenFilterFactory extends AbstractTokenFilterFactory {
```
Comment (jimczi) [inline plugins/analysis-stempel/src/main/java/org/elasticsearch/index/analysis/pl/PolishStopTokenFilterFactory.java]: It seems weird to have a `PolishStopTokenFilterFactory` where the list of stopwords can be overridden in a setting (`stopwords`). Did you add this for the documentation example ? Can we instead use a `stop` filter and set the polish list of stopwords explicitly ?

**Review by jimczi — COMMENTED** [review]
Thanks @cbuescher , I left one comment regarding the `polish_stop` filter which doesn't seem to be needed.

File: plugins/analysis-stempel/src/main/java/org/elasticsearch/index/analysis/pl/PolishStopTokenFilterFactory.java (hunk 1 above)
Comment (cbuescher) [reply plugins/analysis-stempel/src/main/java/org/elasticsearch/index/analysis/pl/PolishStopTokenFilterFactory.java]: The stopword list is in th plugin jar, I tried a few things loading it via the settings that didn't work (one of them the obvious thing trying to load the polish stopwords with `"stopwords" : "_polish_"`  but for that to work I think the stopworks would need to get registered in the Analysis#NAMED_STOP_WORDS map, but we don't seem to have a straight forward way to do that from a plugin). Accessing the "stopwords.txt" file via "stopwards_path" in the plugin jar didn't work either, it tries to load files from the "conf" directory. 
I can change the "polish_stop" to not be configurable, but this is the same we do e.g. in `JapaneseStopTokenFilterFactory` where I got the idea from.

Comment (jimczi) [reply plugins/analysis-stempel/src/main/java/org/elasticsearch/index/analysis/pl/PolishStopTokenFilterFactory.java]: Ok thanks for explaining. However if we add this filter we should have a documentation similar to https://www.elastic.co/guide/en/elasticsearch/plugins/master/analysis-kuromoji-stop.html. I think it's ok if we allow custom list of stopwords in this filter but we need to ensure (I think it's already the case) that `_polish_` can be used to add the default list. It won't work in the default `stop` filter but this shouldn't be an issue.

**Comment by cbuescher** [comment]
@jimczi I added a section about the polish stop filter. Does this look okay to you now?

**Review by jimczi — APPROVED** [review]
Thanks @cbuescher !

**Review by pgomulka — APPROVED** [review]
looks good!

**Pull request merged**